evaluator/builtin.go

```diff
@@ -167,6 +167,24 @@ var Funcs = map[string]Func{
 	ast.GetVar:     {builtinGetVar, 1, 1},
 }
 
+// FuncsCnntBeCnstntFld is for functions that
+// use input parameter ctx or
+// return an uncertain result would not be constant folded
+// the value 0 means nothing
+var FuncsCnntBeCnstntFld = map[string]int{
```

Comment:
I did some fix, and DynamicFuns would be used in package 'expression', so I make it public